karma.conf.js

```diff
@@ -1,51 +1,23 @@
-module.exports = function (config) {
+module.exports = function(config) {
   config.set({
-    frameworks: ['browserify', 'detectBrowsers', 'mocha'],
-    files: [
-      'test/*.js'
-    ],
+    frameworks: ['mocha', 'karma-typescript'],
+    files: ['src/**/*.ts', 'test/**/*.ts'],
     preprocessors: {
-      'test/*.js': ['browserify', 'env']
+      '**/*.ts': ['karma-typescript'],
     },
-    singleRun: true,
-    plugins: [
-      'karma-browserify',
-      'karma-chrome-launcher',
-      'karma-env-preprocessor',
-      'karma-firefox-launcher',
-      'karma-detect-browsers',
-      'karma-mocha'
-    ],
-    browserify: {
-      'transform': [
-        [
-          'babelify',
-          {
-            'presets': [
-              'env'
-            ]
-          }
-        ]
-      ],
-      debug: true
+    plugins: ['karma-mocha', 'karma-typescript', 'karma-chrome-launcher', 'karma-firefox-launcher'],
+    karmaTypescriptConfig: {
+      bundlerOptions: {
+        entrypoints: /\.spec\.ts$/,
+      },
     },
-    envPreprocessor: [
-      'RANDOM_TESTS_REPEAT',
-      'TRAVIS'
-    ],
-    detectBrowsers: {
-      enabled: true,
-      usePhantomJS: false,
-      postDetection: function (availableBrowser) {
-        if (availableBrowser.includes('Chrome')) {
-          return ['ChromeHeadless']
-        }
-
-        var browsers = ['Chrome', 'Firefox']
-        return browsers.filter(function (browser) {
-          return availableBrowser.indexOf(browser) !== -1
-        })
-      }
-    }
+    colors: true,
+    reporters: ['progress', 'karma-typescript'],
+    browsers: ['FirefoxHeadless', 'ChromeHeadless'],
+    singleRun: true,
+    concurrency: Infinity,
+    // Fail after timeout
+    browserDisconnectTimeout: 100000,
+    browserNoActivityTimeout: 100000,
```

Comment:
Could you add some notes on the config you changed here together with some reasoning along?

Reply:
I grabbed some of the sensible defaults that were being used in ethereumjs-vm/karma.conf.js. I was going to bring over the comments indicating what each line does but decided not to for brevity, next time I will. Thanks!